### DISCUSSION TITLE
How do I set Powerdelta depending on the current power consumption?

### DISCUSSION THREAD

**Discussion by Tidle999** [arendst/Tasmota#13049] [Q&A] [answered]
Hello together,

I know there is an example for exactly this in the documentation, however it does not work properly. As soon as the rule is run through and the power consumption increases from 5w to 100w, for example, the power delta does not change. Do you have any idea?

Documentation:

`Backlog PowerDelta 0; Rule1 0; Rule1 5

Rule1
  ON ENERGY#Power>=35 DO Backlog PowerDelta 10; Status 8 BREAK 
  ON ENERGY#Power>=15 DO Backlog PowerDelta 25; Status 8 BREAK 
  ON ENERGY#Power>5 DO Backlog PowerDelta 35; Status 8 BREAK 
  ON ENERGY#Power<=5 DO PowerDelta 100 ENDON

Rule1 1 `

i thought rule example has to be changed from

`Backlog PowerDelta 0; Rule1 0; Rule1 5

Rule1
ON ENERGY#Power>=35 DO Backlog PowerDelta 10; Status 8 BREAK
ON ENERGY#Power>=15 DO Backlog PowerDelta 25; Status 8 BREAK
ON ENERGY#Power>5 DO Backlog PowerDelta 35; Status 8 BREAK
ON ENERGY#Power<=5 DO PowerDelta 100 ENDON

Rule1 1`

to

`Backlog PowerDelta 0; Rule1 0; Rule1 5

Rule1

ON ENERGY#Power<5 DO PowerDelta 100; Status 8 BREAK
ON ENERGY#Power>=5 DO Backlog PowerDelta 35; Status 8 BREAK
ON ENERGY#Power>=15 DO Backlog PowerDelta 25; Status 8 BREAK
ON ENERGY#Power>=35 DO Backlog PowerDelta 10 ENDON

Rule1 1`
but that did not work either. 

Some log where you can see power is above 100w but power delta is wrong. It should be 10 but its 35.

Status 9
13:31:37.856 MQT: stat/tasmota_610E24/STATUS9 = {"StatusPTH":{"PowerDelta":[35,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
13:31:47.246 CMD: Status 8
13:31:47.256 MQT: stat/tasmota_610E24/STATUS8 = {"StatusSNS":{"Time":"2021-09-03T13:31:47","ENERGY":{"TotalStartTime":"2021-08-16T08:53:32","Total":31.100,"Yesterday":2.070,"Today":1.166,"Power":128,"ApparentPower":158,"ReactivePower":92,"Factor":0.81,"Voltage":234,"Current":0.675}}}

**Comment by meingraham** [comment]
Yes, your rule change would definitely not work for the reasons I mentioned in my earlier reply.

The one-shot applies to the rule set (e.g., `Rule1`). Once a trigger in the rule set triggers, the rule set cannot be triggered by the same trigger in that rule set again until a different trigger in the rule set fires (which then sets the one-shot for that trigger).

I developed this rule (and contributed it to the Rule Cookbook). It's been working on several 6.x/7.x devices for a couple of years or more. I haven't upgraded any of my devices recently... because IIABDFI! :wink:

**Reply by Tidle999** [reply]
so it should work like it is described in the documentation? Then it’s a Programming issue with the newer firmware versions? Either the rule is wrong or the Code is wrong. 

as you can see in the logs here:

i am using exactly the same rule as in the documentation. But when power goes from 8w to over 100w. PowerDelta does not change from 35 to 10. 

18:31:34.622 CMD: Rule1
18:31:34.630 MQT: stat/tasmota_610E24/RESULT = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":232,"Free":279,"Rules":"ON ENERGY#Power>=35 DO Backlog PowerDelta 10; Status 8 BREAK    ON ENERGY#Power>=15 DO Backlog PowerDelta 25; Status 8 BREAK    ON ENERGY#Power>5 DO Backlog PowerDelta 35; Status 8 BREAK    ON ENERGY#Power<=5 DO PowerDelta 100 ENDON"}}
18:31:41.170 CMD: Status 8
18:31:41.181 MQT: stat/tasmota_610E24/STATUS8 = {"StatusSNS":{"Time":"2021-09-03T18:31:41","ENERGY":{"TotalStartTime":"2021-08-16T08:53:32","Total":31.502,"Yesterday":2.070,"Today":1.567,"Power":127,"ApparentPower":156,"ReactivePower":91,"Factor":0.82,"Voltage":231,"Current":0.678}}}
18:31:44.849 CMD: Status 9
18:31:44.855 MQT: stat/tasmota_610E24/STATUS9 = {"StatusPTH":{"PowerDelta":[35,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}

**Reply by Tidle999** [reply]
Does really no one has an idea? What about disabling the one-shot feature? 
Rule1 4

**Comment by barbudor** [comment, marked as answer]
I've done something like that in the past using IF to avoid the problem
Let me think about it

**Reply by barbudor** [reply]
I don't have IF in my power socket but you can look at what I've done for adapting DeepSleepTime to battery level here :
https://tasmota.github.io/docs/DeepSleep/#rules-to-adapt-deepsleep-period-to-battery-level
